Add Playwright check type comparison (#80)
checkly/docs: detect/synthetic-monitoring/playwright-monitoring-comparison.mdx

---

> title: 'Playwright Monitoring Comparison'

Reviewer: ```suggestion
title: 'Playwright Monitoring Options'
```

I feel that the page first explains the options, helps make a choice, then compares with the table? I'm not too settled on this one; it's tricky for sure.

description: 'Compare the features of the different Playwright-based checks'
sidebarTitle: 'Playwright Feature Comparison'

---

import { checkDurations } from '/snippets/check-comparison.mdx'

[MultiStep Checks](/detect/synthetic-monitoring/multistep-checks/overview), [Browser Checks](/detect/synthetic-monitoring/browser-checks/overview), and [Playwright Check Suites](/detect/synthetic-monitoring/playwright-checks/overview) use Playwright for synthetic monitoring. The different check types differ in terms of capabilities, flexibility, and use cases.

Checkly's strengths are that you can log in to the Checkly web application, write Playwright code in the built-in editor, and start monitoring your APIs and sites. **MultiStep and Browser checks are the perfect tools to get you off the ground quickly.** Don't worry about where to store the monitoring code, how to configure advanced settings, or when to update dependencies. Checkly takes care of all of this.

> However, when your monitoring setup needs to match scaling and growing project requirements, you'll quickly find yourself looking for ways to version-control your checks with [the Monitoring as Code approach \(MaC\)](/concepts/monitoring-as-code).

Reviewer: ```suggestion
However, when your monitoring setup needs to scale with your growing project requirements, you'll quickly look for ways to version-control your checks using [the Monitoring as Code approach \(MaC\)](/concepts/monitoring-as-code).
```

All Playwright-based checks can be created, updated and configured as code. But, when should you use which?

## When to use MultiStep Checks

[MultiStep Checks](/detect/synthetic-monitoring/multistep-checks/overview) enable you to monitor complex API workflows with sequential requests using the familiar Playwright API. The check run is based on a single `spec.ts` file, limited to making HTTP requests (there's no browser available) and **the max check run duration is capped at { checkDurations.multiStep }**.

## When to use Browser Checks

[Browser Checks](/detect/synthetic-monitoring/browser-checks/overview) let you monitor end-to-end user flows and realistic interactions with automated browser testing using Playwright. The check run is based on a single `spec.ts` file, supports the headless Chromium/Chrome browser and **the max check run duration is capped at { checkDurations.browser }**.

## When to use Playwright Check Suites

> 

Reviewer: ```suggestion
Choose Playwright Check Suites when you need cross-browser support, custom dependencies, or when you already have an existing Playwright test suite that includes critical flows that should become checks.
```

MultiStep and Browser Checks are single units based on a single `spec.ts` file.

[Playwright Check Suites](/detect/synthetic-monitoring/playwright-checks/overview), on the other hand, let you take your entire existing Playwright test suite, and turn it into globally distributed monitoring. Check Suites support multiple spec files, advanced global configuration, multiple Playwright projects, tags for selective monitoring, storage state, custom dependencies, and more.

<Tip>
Playwright Check Suites are the native way to run your Playwright project in production.
</Tip>

## Feature comparison

| Feature                                       | Playwright Check Suites       | Browser Check                   | MultiStep Check                 |
|-----------------------------------------------|-------------------------------|---------------------------------|---------------------------------|
| Supports *Monitoring as Code*                 | Yes                           | Yes                             | Yes                             |

> | Supported test formats                        | API and Browser               | API and Browser                 | API only                        |

Reviewer: ```suggestion
| [Test fixtures ](https://playwright.dev/docs/test-fixtures)                     | API and Browser               | API and Browser                 | API only                        |
```

Would it make sense to make this Playwright named? As In Test Fixtures

| Multiple spec files                           | Yes                           | No                              | No                              |
| Flexible test selection via tags and projects | Yes                           | No                              | No                              |

> | Browser compatibility                         | Chromium, WebKit, Firefox     | Chromium, Chrome                | —                               |

Author: @MariadeAnton I'll reject this because you can emulate mobile devices in your Browser checks. :)

| Storage state support                         | Yes                           | No                              | No                              |

> | Support for multiple Playwright projects      | Yes                           | No                              | No                              |

Reviewer: ```suggestion
| Multiple Playwright Projects    | Yes                           | No                              | No                              |
```

---
title: 'Playwright Monitoring Options'
description: 'Compare the features of the different Playwright-based checks'

> sidebarTitle: 'Playwright Feature Comparison'

Reviewer: ```suggestion
sidebarTitle: 'Playwright Monitoring Options'
```
should these agree with the title as closely as possible?
